pkg/token/token.go

```diff
@@ -122,6 +122,26 @@ type getCallerIdentityWrapper struct {
 	} `json:"GetCallerIdentityResponse"`
 }
 
+// this structure is saved to cache for future invocations
+type cachedToken struct {
+	ExpiryDate time.Time
+	Token      string
+}
+
+func getCacheFile(clusterID string, roleARN string) (string, error) {
+	cacheDir := os.Getenv("XDG_CACHE_HOME")
```

Comment:
I think its reasonable to use a dependency to help us use customary default locations for placing the cache file (or just look at the locations they use and avoid the dependency).

Reply:
The main reason this would require a dependency (or at least one would be preferred) is because this is usually done by choosing the paths to include at compile time. If there's no preference I'll just pick one and update accordingly.